File: backend/app/schemas/audit.py
"""Explicit audit API schemas."""

from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field


class AuditEventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    event_type: str
    event_version: int
    timestamp: datetime
    action: str
    result: str
    correlation_id: str
    actor_user_id: str | None
    actor_session_id: str | None
    target_type: str | None
    target_id: str | None
    reason: str | None
    permission: str | None
    source_ip: str | None
    user_agent: str | None
    metadata: dict = Field(validation_alias="event_metadata", serialization_alias="metadata")
    created_at: datetime


class AuditEventPage(BaseModel):
    items: list[AuditEventResponse]
    next_cursor: str | None

File: apps/operator/src/types/audit.ts
export type AuditResult = 'SUCCESS' | 'FAILURE';

export interface AuditEvent {
  id: string;
  event_type: string;
  event_version: number;
  timestamp: string;
  action: string;
  result: AuditResult | string;
  correlation_id: string;
  actor_user_id?: string | null;
  actor_session_id?: string | null;
  target_type?: string | null;
  target_id?: string | null;
  reason?: string | null;
  permission?: string | null;
  source_ip?: string | null;
  user_agent?: string | null;
  metadata?: Record<string, unknown>;
  created_at: string;
}

export interface AuditEventPage {
  items: AuditEvent[];
  next_cursor?: string | null;
}

export interface AuditFilterParams {
  event_type?: string;
  result?: string;
  actor_id?: string;
  target_type?: string;
  target_id?: string;
  permission?: string;
  date_from?: string;
  date_to?: string;
  cursor?: string;
  limit?: number;
}
